browser/ui/webui/brave_education/education_page_handler_unitttest.cc

```diff
@@ -0,0 +1,164 @@
+ protected:
+  void SetUp() override {
+    TestingProfile::Builder builder;
+    profile_ = builder.Build();
+  }
+
+  mojo::Remote<brave_browser_command::mojom::BraveBrowserCommandHandler>&
+  CreateHandler(Profile* profile = nullptr) {
+    auto delegate = std::make_unique<TestDelegate>(base::BindLambdaForTesting(
+        [this](std::string action) { actions_.push_back(std::move(action)); }));
+
+    if (!profile) {
+      profile = profile_.get();
+    }
+
+    std::vector<brave_browser_command::mojom::Command> supported_commands = {};
+    supported_commands.insert(
+        supported_commands.end(),
+        {brave_browser_command::mojom::Command::kOpenRewardsOnboarding,
+         brave_browser_command::mojom::Command::kOpenWalletOnboarding,
+         brave_browser_command::mojom::Command::kOpenVPNOnboarding,
+         brave_browser_command::mojom::Command::kOpenAIChat});
+
+    page_handler_ = std::make_unique<BraveBrowserCommandHandler>(
+        remote_.BindNewPipeAndPassReceiver(), profile, supported_commands,
+        std::move(delegate));
+
+    return remote_;
+  }
+
+  Profile& profile() { return *profile_; }
+
+  const std::vector<std::string>& actions() { return actions_; }
+
+ private:
+  content::BrowserTaskEnvironment task_environment_;
+  ScopedTestingLocalState local_state_{TestingBrowserProcess::GetGlobal()};
+  mojo::Remote<brave_browser_command::mojom::BraveBrowserCommandHandler>
+      remote_;
+  std::unique_ptr<Profile> profile_;
+  std::unique_ptr<BraveBrowserCommandHandler> page_handler_;
+  std::vector<std::string> actions_;
+};
+
+TEST_F(EducationPageHandlerTest, BasicCommandsExecuted) {
+  auto& handler = CreateHandler();
+
+  base::test::TestFuture<bool> future;
+
+  handler->ExecuteCommand(
+      brave_browser_command::mojom::Command::kOpenWalletOnboarding,
+      base::DoNothing());
+  handler->ExecuteCommand(
+      brave_browser_command::mojom::Command::kOpenRewardsOnboarding,
+      future.GetCallback());
+
+  ASSERT_TRUE(future.Get());
+  EXPECT_EQ(actions()[0], "open-url: chrome://wallet/");
+  EXPECT_EQ(actions()[1], "open-rewards-panel");
+}
+
+TEST_F(EducationPageHandlerTest, VPNCommandsExecuted) {
+  auto& handler = CreateHandler();
+  base::test::TestFuture<bool> future;
+
+  handler->ExecuteCommand(
+      brave_browser_command::mojom::Command::kOpenVPNOnboarding,
+      future.GetCallback());
+
+#if BUILDFLAG(ENABLE_BRAVE_VPN)
+  ASSERT_TRUE(future.Get());
+  EXPECT_EQ(actions()[0], "open-vpn-panel");
+#else
+  ASSERT_FALSE(future.Get());
+  EXPECT_TRUE(actions().empty());
+#endif
+}
+
+TEST_F(EducationPageHandlerTest, ChatCommandsExecuted) {
+  auto& handler = CreateHandler();
+  base::test::TestFuture<bool> future;
+
+  handler->ExecuteCommand(brave_browser_command::mojom::Command::kOpenAIChat,
+                          future.GetCallback());
+
+  ASSERT_TRUE(future.Get());
+  EXPECT_EQ(actions()[0], "open-ai-chat");
+}
+
+// TODO(bsclifton): this currently fails. The web UI won't load
```

Comment:
?

Reply:
according to https://github.com/brave/brave-core/pull/26154/files#r1915447183 it's not allowed in incognito so should fail, but I'm not sure why

Reply:
Before we merge, I do need to fix this. This was passing before - but the one handler was split to two. Need to make some changes. Will do this ASAP!

Reply:
Fixed in https://github.com/brave/brave-core/pull/26154/commits/f32cc1dcb3dd7af378fee21b621b7a014ac71095